drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -524,312 +500,327 @@
-          Integrity Protected Option-Type header MUST act as a Validator.
-          Otherwise, an attacker could modify the IOAM header along the path and
-          change the actions a node performs. Examples:
+        <t>The encapsulating node writes the ICV in the corresponding field of
+        the IOAM Integrity Protection header. Reminder: it MUST always be a
+        16-octet ICV.</t>
 
-          <list style="bullets">
-            <t>For an Integrity Protected Trace Option-Type (Pre-allocated or
-            Incremental), each transit node MUST act as a Validator, if either
-            the IOAM Loopback or Active mode is used.</t>
+        <section anchor="SelectHdrFields" title="Selection of header fields">
+          <t>The main objective of the Integrity Protection Method defined in
+          this document is to provide integrity protection of IOAM-Data-Fields.
+          However, some Option-Type header fields are crucial for
+          IOAM-Data-Fields. Without them, IOAM-Data-Fields are meaningless.
+          Therefore, the integrity of such header fields MUST be protected too.
+          </t>
 
-            <t>For an Integrity Protected DEX Option-Type, each transit node
-            MUST act as a Validator.</t>
-          </list>
+          <t>For Trace Option-Types, here is the list of header fields that
+          participate (in that order) in the integrity protection of
+          IOAM-Data-Fields:
+            <list style="numbers">
+              <t>Namespace-ID</t>
+              <t>IOAM-Trace-Type</t>
+            </list>
+
+          The NodeLen field is not included in the list because it can be
+          deduced from the IOAM-Trace-Type field. Other header fields that are
+          not included in the list are either mutable or only useful for
+          processing Trace Option-Types (i.e., they don't provide context or
+          meaning to IOAM-Data-Fields).</t>
+
+          <t>For a POT Option-Type, here is the list of header fields that
+          participate (in that order) in the integrity protection of
+          IOAM-Data-Fields:
+            <list style="numbers">
+              <t>Namespace-ID</t>
+              <t>IOAM-POT-Type</t>
+            </list>
+
+          Other header fields that are not included in the list are either
+          mutable or only useful for processing a POT Option-Type (i.e., they
+          don't provide context or meaning to IOAM-Data-Fields).</t>
+
+          <t>For a E2E Option-Type, here is the list of header fields that
+          participate (in that order) in the integrity protection of
+          IOAM-Data-Fields:
+            <list style="numbers">
+              <t>Namespace-ID</t>
+              <t>IOAM-E2E-Type</t>
+            </list>
+
+          Those are all the header fields defined for a E2E Option-Type.</t>
+
+          <t>A document defining a new IOAM Integrity Protected Option-Type that
```

Comment:
s/A document defining a new IOAM Integrity Protected Option-Type that intends to/New IOAM Integrity Protected Option-Types that intend to/